checker/tests/lock/ChapterExamples.java

```diff
@@ -0,0 +1,859 @@
+// This test contains the sample code from the Lock Checker manual chapter
+// modified to fit testing instead of illustrative purposes,
+// and contains other miscellaneous Lock Checker testing.
+import org.checkerframework.checker.lock.qual.*;
+import org.checkerframework.checker.nullness.qual.*;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.Deterministic;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+import org.checkerframework.framework.qual.ImplicitFor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedPrimitiveType;
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.*;
+
+class ChapterExamples {
+    // This code crashed when there was a bug before issue 524 was fixed.
+    // An attempt to take the LUB between 'val' in the store with type 'long'
+    // and 'val' in another store with type 'none' resulted in a crash.
+    private void foo (boolean b, int a) {
+        if (b) {
+          if (a == 0) {
+            boolean val = false;
+          } else if (a == 1) {
+            int val = 0;
+          } else if (a == 2) {
+            long val = 0;
+          } else if (a == 3) {
+          }
+        } else {
+          if (true) {
+          }
+        }
+    }
+
+
+    private abstract class Values<V> extends AbstractCollection<V> {
+        public <T> T[] toArray(T[] a) {
+            Collection<V> c = new ArrayList<V>(size());
+            for (Iterator<V> i = iterator(); i.hasNext(); )
+                c.add(i.next());
+            return c.toArray(a);
+        }
+    }
+
+
+
+    // Keep these as test cases!
+    <T> T method1(T t, boolean b) {
+        //(return.type.incompatible)
```

Comment:
Should this start with "//:: error:" ? The comment on top says that this file should be a test case.